test/unit/graphql_metrics_integration_test.rb

```diff
@@ -0,0 +1,995 @@
+    }, {
+      :field_name => "post",
+      :return_type_name => "Post",
+      :parent_type_name => "PostCreatePayload",
+      :deprecated => false,
+      :path => ["postCreate", "post"],
+      :resolver_timings => [{
+        :start_time_offset => SomeNumber.new(at_least: SMALL_NONZERO_NUMBER),
+        :duration => SomeNumber.new(at_least: SMALL_NONZERO_NUMBER),
+      }],
+      :lazy_resolver_timings => nil
+    }, {
+      :field_name => "postCreate",
+      :return_type_name => "PostCreatePayload",
+      :parent_type_name => "MutationRoot",
+      :deprecated => false,
+      :path => ["postCreate"],
+      :resolver_timings => [{
+        :start_time_offset => SomeNumber.new(at_least: SMALL_NONZERO_NUMBER),
+        :duration => SomeNumber.new(at_least: SMALL_NONZERO_NUMBER),
+      }],
+      :lazy_resolver_timings => nil
+    }]
+
+    assert_equal(
+      expected_fields,
+      actual_fields,
+      Diffy::Diff.new(JSON.pretty_generate(expected_fields), JSON.pretty_generate(actual_fields))
+    )
+
+    expected_arguments = [{
+      :argument_name => "post",
+      :argument_type_name => "PostInput",
+      :parent_field_name => "postCreate",
+      :parent_field_type_name => "MutationRoot",
+      :default_used => false,
+      :value_is_null => false,
+      :value => SomeArgumentValue.new,
+    }, {
+      :argument_name => "title",
+      :argument_type_name => "String",
+      :parent_field_name => "postCreate",
+      :parent_field_type_name => "MutationRoot",
+      :default_used => false,
+      :value_is_null => false,
+      :value => SomeArgumentValue.new,
+    }, {
+      :argument_name => "body",
+      :argument_type_name => "String",
+      :parent_field_name => "postCreate",
+      :parent_field_type_name => "MutationRoot",
+      :default_used => false,
+      :value_is_null => false,
+      :value => SomeArgumentValue.new,
+    }, {
+      :argument_name => "embeddedTags",
+      :argument_type_name => "TagInput",
+      :parent_field_name => "postCreate",
+      :parent_field_type_name => "MutationRoot",
+      :default_used => false,
+      :value_is_null => false,
+      :value => SomeArgumentValue.new,
+    }, {
+      :argument_name => "handle",
+      :argument_type_name => "String",
+      :parent_field_name => "postCreate",
+      :parent_field_type_name => "MutationRoot",
+      :default_used => false,
+      :value_is_null => false,
+      :value => SomeArgumentValue.new,
+    }, {
+      :argument_name => "displayName",
+      :argument_type_name => "String",
+      :parent_field_name => "postCreate",
+      :parent_field_type_name => "MutationRoot",
+      :default_used => false,
+      :value_is_null => false,
+      :value => SomeArgumentValue.new,
+    }]
+
+    assert_equal(
+      expected_arguments,
+      actual_arguments,
+      Diffy::Diff.new(JSON.pretty_generate(expected_arguments), JSON.pretty_generate(actual_arguments))
+    )
+  end
+
+  class Schema2 < GraphQL::Schema
```

Comment:
What's the difference here? Maybe name them more semantically? I'd also probably just define this up top too

Reply:
I'll rename them. One schema gets both static and runtime metrics, and one (Schema2) gets only static metrics to demonstrate this is all possible if someone wants usage but not timings.